[FEATURE] Afficher la nouvelle double mire lors d'une connexion via un partenaire oidc (PIX-4988).

AndreiaPena opened the pull request:
## :unicorn: Problème
L'actuel scénario de réconciliation de compte n'est plus pertinent. Il permet entre autres de mauvaises réconciliation, comme un élève sco connecté sur Pix dont le parent va se connecter via Pôle Emploi en commençant une campagne.
Les deux comptes sont alors réconcilés.

## :robot: Solution
Désormais dans le nouveau scénario, les utilisateurs déjà authentifié à Pix qui souhaitent se connecter via un partenaire oidc seront déconnectés et accèderont à une nouvelle double mire Pix.

Cette nouvelle page propose deux solutions : 
- Si la personne n'a pas de compte Pix, il n'a plus qu'à valider les cgu et un compte Pix sera créé
- Si la personne à déjà un compte, on invite à se connecter via email et mot de passe. 

## :rainbow: Remarques
Cette nouvelle double mire est sous feature toggle, celui-ci à du être renommé dans ce ticket.

Une maquette en version mobile est disponible mais elle sera implémenté ultérieurement.

Cette PR ne s'occupe que de la partie front. 
Pour le cas d'un utilisateur sans compte qui doit valider les cgu, c'est raccordé au back existant. 
Pour l'autre solution, rien n'a été fait coté back

## :100: Pour tester
- Si test en local, s'assurer d'avoir la variable du feature toggle : `FT_SSO_ACCOUNT_RECONCILIATION=true`
- Taper l'url `/connexion/pole-emploi` ou cliquer sur le bouton PE présent dans `/connexion`
- se connecter via PE
- Constater que l'on accède à la nouvelle double mire
- Tester la partie de gauche
- Faire les tests d'erreurs : ne pas valider les cgu et cliquer sur le bouton (message d'erreur), authenticationKey expirée...
- Faire le cas passant et constater que l'on accède au user-dashboard avec création de compte
- ( ⚠️ comme c'est raccordé au back existant, coté PE, ce scénario va faire la réconciliation si compte précédemment connecté alors que c'est pas le but. Back à changer)
- Tester la partie droite juste en ignorant des champs, en mettant un email invalide
- L'action du bouton ne donne rien

AndreiaPena committed:
refactor(api): Use camel case on feature toggle

AndreiaPena committed:
refactor(mon-pix): Use camel case on feature toggle

AndreiaPena committed:
feat(mon-pix): add new form for oidc authentication

AndreiaPena committed:
refactor(mon-pix): move cgu translations in common object

pix-service commented:
I'm deploying this PR to these urls:

- App (.fr): https://app-pr4730.review.pix.fr
- App (.org): https://app-pr4730.review.pix.org
- Orga: https://orga-pr4730.review.pix.fr
- Certif: https://certif-pr4730.review.pix.fr
- Admin: https://admin-pr4730.review.pix.fr
- API: https://api-pr4730.review.pix.fr/api/

 Please check it out!

AndreiaPena committed:
feat(mon-pix): add register form on new page

AndreiaPena commented:
**Suggestion** : On peut ajouter ce nouveau composant dans le formulaire d'exemple 🙂 https://ui-pr232.review.pix.fr/?path=/docs/form-example--form

AndreiaPena committed:
feat(mon-pix): add login form

AndreiaPena committed:
style(mon-pix): adapt css for mobile and tablet devices

igorissen commented on mon-pix/app/components/authentication/login-or-register-oidc.hbs:
```diff
@@ -0,0 +1,88 @@
+<PixBackgroundHeader>
```
Le mobile sera pris en compte dans une PR suivante, par contre est-ce que c'est possible de réglé ce souci en tablette (mode paysage - 1024px) ?

![Screenshot 2022-08-04 at 09 09 50](https://user-images.githubusercontent.com/6919604/182786822-764bc176-ea80-4afe-9fd7-469fce59f764.png)

AndreiaPena replied:
corrigé !

igorissen commented on mon-pix/tests/integration/components/authentication/login-or-register-oidc_test.js:
```diff
@@ -0,0 +1,77 @@
+      );
+
+      // then
+      expect(
```
Est-ce qu'il ne faudrait pas ajouter une assertion qui vérifie bien qu'il y a le terme `Pôle Emploi` ?

AndreiaPena replied:
fait !

yschoueri commented on mon-pix/app/components/authentication/login-or-register-oidc.hbs:
```diff
@@ -0,0 +1,88 @@
+        </PixButton>
+      </div>
+
+      <div class="login-or-register-oidc-form__divider"></div>
```
chipotage: on pourrait utiliser une `<hr>` avec un transform pour la rendre vertical

AndreiaPena replied:
Avec la maquette mobile, ce divider aura un "OU" entre deux lignes. On aura besoin de cette div du coup

yschoueri commented on mon-pix/app/components/authentication/login-or-register-oidc.js:
```diff
@@ -0,0 +1,102 @@
+  @tracked emailValidationMessage = null;
+
+  get identityProviderOrganizationName() {
+    return IdentityProviders[this.args.identityProviderSlug].organizationName;
```
🎉 finalement il a bien servi ce organization name

yschoueri commented on mon-pix/app/components/authentication/login-or-register-oidc.js:
```diff
@@ -0,0 +1,102 @@
+  }
+
+  get dataProtectionPolicyUrl() {
+    const currentLanguage = this.intl.t('current-lang');
+    if (currentLanguage === 'en') {
```
chipotage: vu que c'est fait a 2 reprises on pourrait faire un `get currentLanguage()`
```suggestion
    if (this.currentLanguage === 'en') {
```

AndreiaPena replied:
done

yschoueri commented on mon-pix/app/components/authentication/login-or-register-oidc.js:
```diff
@@ -0,0 +1,102 @@
+    if (this.isTermsOfServiceValidated) {
+      this.errorMessage = null;
+      try {
+        await this.session.authenticate('authenticator:oidc', {
+          authenticationKey: this.args.authenticationKey,
+          identityProviderSlug: this.args.identityProviderSlug,
+        });
```
❤️  quand on voit que la mutualisation facilite la vie

yschoueri commented on mon-pix/app/components/authentication/login-or-register-oidc.hbs:
```diff
@@ -0,0 +1,88 @@
+          }}
+        </p>
+        <div class="login-or-register-oidc-form__cgu-container">
+          <Input
```
On ne pourrait pas utiliser le pixcheckbox ici, ou tu feras la modif dans ta prochaine PR?

AndreiaPena replied:
Oui prochaine PR je préfère :D

yschoueri commented on mon-pix/app/styles/components/authentication/_login-or-register-oidc.scss:
```diff
@@ -0,0 +1,128 @@
+
+  &__title {
+    color: $pix-neutral-90;
+    font-family: $font-open-sans;
+    font-weight: 300;
+    font-size: 46px;
+    line-height: 60px;
+    text-align: center;
```
est-ce qu'il n'y a pas moyen d'utiliser les classes qui existe deja du design system ?  cette question s'applique pour tous les elements de la page

AndreiaPena replied:
j'ai juste pas l'habitude de ces nouvelles classes 😅 où est-ce qu'on les voit ? On a une liste de ces classes ? :)

yschoueri replied:
pour le moment, le seul moyen c'est d'ouvrir le css 🙈 j'avoue ce serait bien de l'ajouter a la doc pix ui comme on n'a fait pour les palettes

yschoueri replied:
mais tu peux te baser sur figma et le nom des classes et mixin n'est pas tres differente

yschoueri commented on mon-pix/app/styles/components/authentication/_login-or-register-oidc.scss:
```diff
@@ -0,0 +1,128 @@
+    flex-direction: column;
+    align-items: center;
+
+    @include device-is('desktop') {
```
j'irai dans l'autre sens, parce que par defaut flex-direction est row donc tu n'aurais pas besoin de l'ecrire et dans le cas ou c'est mobile, on change la direction

AndreiaPena replied:
j'essaye d'être mobile first à chaque fois c'est pour ça :/

yschoueri replied:
oui j'ai compris quand j'ai vu ca mais vu que l'appli n'est vraiment pas mobile first, est-ce necessaire ? 🤷‍♀️

AndreiaPena replied:
Ouais de toute façon si on regarde le code, c'est un peu à l'appréciation de chacun 😅 
Ici si c'est juste pour pas devoir changer le flex-direction à nouveau, je préfère ne rien changer. Comme ça tout mon fichier est iso, seul le device qui change est le desktop, c'est plus clair :)

yschoueri commented on mon-pix/tests/integration/components/authentication/login-or-register-oidc_test.js:
```diff
@@ -0,0 +1,77 @@
+    // then
+    expect(
+      screen.getByRole('heading', {
+        name: this.intl.t('pages.login-or-register-oidc.title'),
```
je me demande si on a vraiment besoin d'utiliser la cle de traduction et pas tout de suite mettre le français ? nos tests ne sont pas lancé en français ?

AndreiaPena replied:
A choisir, je préfère tester la clé de trad justement parce qu'on à plusieurs langues sur cette app et qu'il est plus intéressant de tester que la bonne clé de traduction soit bien à l'endroit attendu.
Laura à d'ailleurs fait une doc Ember dans ce sens :) 
https://github.com/1024pix/pix/pull/2773

yschoueri replied:
ben du coup dans tes autres tests tu pourrais faire pareil

AndreiaPena replied:
comme j'ai dupliqué les autres tests, ouais c'est en dur. Je change ça

yschoueri commented on mon-pix/translations/en.json:
```diff
@@ -916,6 +925,26 @@
       }
     },
+    "login-or-register-oidc": {
+      "title": "Créez votre compte Pix",
```
On n'a pas deja les trad ?

AndreiaPena replied:
Non 😅

yschoueri replied:
tu veux que je fasse une premiere passe ?

AndreiaPena replied:
comme c'est sous feature toggle, y'a pas d'urgence. Je préfère merge ce ticket au plus vite :)

AndreiaPena committed:
refactor(mon-pix): provide text on empty translation to fix ember-intl warning

[ember-intl] "pages.terms-of-service-oidc.message" ICU argument mismatch: "en": "identityProviderOrganizationName"

AndreiaPena committed:
refactor(mon-pix): create currentLanguage function in component
